awx/playbooks/action_plugins/insights.py

```diff
@@ -34,24 +35,65 @@ def write_version(self, proj_path, etag):
         with open(file_path, 'w') as f:
             f.write(etag)
 
+    def _obtain_auth_token(self, oidc_endpoint, client_id, client_secret, result):
+        main_url = urljoin(oidc_endpoint, '/.well-known/openid-configuration')
+        response = requests.get(url=main_url, headers={'Accept': 'application/json'})
+        if response.status_code != 200:
+            result['failed'] = True
+            result['msg'] = 'Expected {} to return a status code of 200 but returned status code "{}" instead with content "{}".'.format(
+                main_url, response.status_code, response.content
+            )
+            return result
+
+        auth_url = response.json().get('token_endpoint', None)
+        data = {
+            'grant_type': 'client_credentials',
+            'scope': 'api.console',
```

Comment:
We will be using this oauth token for _both_ inventory updates and project updates. Hopefully this scope gives us all the access we need need to accomplish both those tasks.